### DISCUSSION TITLE
How does WinpkFilter compare to WinDivert in terms of performance and features?

### DISCUSSION THREAD

**Discussion by sn99** [wiresock/ndisapi#21] [General]
Basically the title. I was considering the use of [WinDivert](https://github.com/basil00/Divert) in one of my projects for monitoring network flow when I came across https://github.com/basil00/Divert/issues/52. I understand that `ndisapi` is faster but could not understand the reason as well as how well would it compare to in the long term in case I go beyond just monitoring network.

**Comment by sn99** [comment]
Idk if this is the correct place to ask @wiresock

**Comment by wiresock** [comment]
I haven't conducted comparison tests between WinDivert and WinpkFilter, but I've been using WinpkFilter for various personal and commercial projects. It also has an impressive clientele, including ZScaler, Intel, Diebold, and others, so I must be mindful of its performance impact. For instance, I've recently made modifications to enhance its performance on 10 Gbps networks. You can find more information on this topic here

In general, WinpkFilter offers a wide range of capabilities beyond network monitoring. You can explore the examples section for more information. If you have specific requirements, feel free to ask here, and I'll do my best to assist you.

**Comment by sn99** [comment]
Hmmm, I will look more into it performance and limited resources are my main concern for now. I will ask more questions is I get struck. Thanks for the reply, I will find more narrowed questions related to performance impact.
